pkg/gatherers/clusterconfig/gather_nodes_test.go

```diff
@@ -0,0 +1,199 @@
+package clusterconfig
+
+import (
+	"context"
+	"testing"
+
+	"github.com/openshift/insights-operator/pkg/utils/anonymize"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/client-go/kubernetes/fake"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+func Test_gatherNodes(t *testing.T) {
+	tests := []struct {
+		name          string
+		nodes         *corev1.NodeList
+		wantRecords   []record.Record
+		wantErrsCount int
+	}{
+		{
+			name: "successful retrieval nodes",
+			nodes: &corev1.NodeList{
+				Items: []corev1.Node{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "node1",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "node2",
+						},
+					},
+				},
+			},
+			wantRecords: []record.Record{
+				{
+					Name: "config/node/node1",
+					Item: record.ResourceMarshaller{Resource: &corev1.Node{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "node1",
+						},
+					}},
+				},
+				{
+					Name: "config/node/node2",
+					Item: record.ResourceMarshaller{Resource: &corev1.Node{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "node2",
+						},
+					}},
+				},
+			},
+			wantErrsCount: 0,
+		},
+		{
+			name:          "nodes not found",
+			nodes:         &corev1.NodeList{},
+			wantRecords:   []record.Record{},
+			wantErrsCount: 0,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			coreClient := fake.NewSimpleClientset(tt.nodes)
+			records, errs := gatherNodes(context.TODO(), coreClient.CoreV1())
```

Comment:
In the other unit tests you use `context.Background()`, which makes more sense IMO.